.github/workflows/publish-javadoc.yml

```diff
@@ -0,0 +1,25 @@
+name: Publish PathPlannerLib Javadoc
+
+on:
+  workflow_dispatch
+
+jobs:
+  build-and-deploy:
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@v3
+      - name: Build PathPlannerLib
+        working-directory: pathplannerlib
+        run: |
+            ./gradlew build -PreleaseMode
+            ./gradlew publish -PreleaseMode
+      - name: Deploy generated Javadoc to GitHub Pages
+        uses: s0/git-publish-subdir-action@v2.6.0
+        env:
+          REPO: git@github.com:jasonli0616/pathplanner.git #change to git@github.com:mjansen4857/pathplanner.git before merge
+          FOLDER: pathplannerlib/build/docs/javadoc
+          BRANCH: gh-pages
+          TARGET_DIR: docs/java
+          SSH_PRIVATE_KEY: ${{ secrets.DEPLOY_PRIVATE_KEY }}
+          MESSAGE: Publish docs from PathPlannerLib version ${{ github.event.release.tag_name }}
```

Comment:
Last thing I can think of is this version label probably won't work anymore since its a manual dispatch, and it would be the GUI version anyway. A general "Publish PathPlannerLib java docs" commit will be fine I think.

Other than that, looks good to me.